agenda-webapps/src/main/webapp/vue-app/agenda-connectors/agendaOfficeConnector.js

```diff
@@ -0,0 +1,212 @@
+              .then(gEvents => resolve(gEvents))
+              .catch(e => {
+                this.loadingCallback(this, false);
+                reject(e);
+              })
+            ,(error) => reject(error)
+          );
+        }*/
+      });
+    } else {
+      return Promise.resolve(null);
+    }
+  },
+};
+
+
+/**
+ * Load Office Connector API javascript and prepare user authentication and
+ * authorization process
+ *
+ * @param {Object}
+ *          connector Office Connector SPI
+ * @returns {void}
+ */
+function initOfficeConnector(connector) {
+  // Called when the signed in status changes, to update the UI
+  // appropriately. After a sign-in, the API is called.
+  connector.loadingCallback(connector, true);
+  window.require(['https://alcdn.msauth.net/browser/2.8.0/js/msal-browser.min.js'], (msal) => {
+    const officeApi = new msal.PublicClientApplication(connector.config);
+    connector.officeApi = officeApi;
+
+    if (officeApi.getAllAccounts().length > 0) {
+      const currentUser = connector.officeApi.getAllAccounts()[0];
+      if(currentUser) {
+        connector.isSignedIn = true;
+        connector.connectionStatusChangedCallback(connector, {
+          user: currentUser.username,
+          id: currentUser.localAccountId,
+        });
+      }
+    }
+    connector.loadingCallback(connector, false);
+  }, (error) => {
+    connector.canConnect = false;
+    connector.loadingCallback(connector, false);
+    console.error('Error retrieving Office API Javascript', error);
+  });
+}
+
+
+/**
+ * @param {Object}
+ *          connector Google Connector SPI
+ * @param {Date}
+ *          periodStartDate Start date of period of events to retrieve
+ * @param {Date}
+ *          periodEndDate End date of period of events to retrieve
+ * @returns {Promise} a promise with list of Google events
+ */
+function retrieveEvents(connector, periodStartDate, periodEndDate) {
+  //const currentUser = connector.officeApi.getAllAccounts()[0];
+  let params = {
+    startDateTime: periodStartDate,
+    endDateTime: periodEndDate,
+  };
+
+  params = $.param(params, true);
+  return connector.getTokenPopup(connector.CalendarRequest).then(calendarToken => {
+    return callOfficeApi(`${connector.graphConfig.graphCalendarEventsEndpoint}${params}`,
+      calendarToken.accessToken,
+      (response) => {
+        const events = response.value;
+        events.forEach(event => {
+          event.allDay = !!event.isAllDay;
+          const StartDate = event.start.dateTime && new Date(`${event.start.dateTime}Z`);
+          const EndDate = event.end.dateTime && new Date(`${event.end.dateTime}Z`);
+          event.start = StartDate || event.start.date;
+          event.end = EndDate  || event.start.date;
+          event.summary = event.subject;
+          event.location = event.location.displayName;
+          event.type = 'remoteEvent';
+          event.color = '#FFFFFF';
```

Comment:
hardcoded?

Reply:
Those are the values we want (type and color) for remote events.